src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exception.DeleteOutOfRangeException;
+import duke.exception.DoneOutOfRangeException;
+import duke.exception.EmptyDeadlineException;
+import duke.exception.EmptyEventException;
+import duke.exception.EmptyTodoException;
+import duke.exception.MissingDeadlineDateException;
+import duke.exception.MissingDeleteArgumentException;
+import duke.exception.MissingDoneArgumentException;
+import duke.exception.MissingEventDateException;
+import duke.exception.MissingFindArgumentException;
+import duke.exception.UnknownCommandException;
+
+/**
+ * Duke Bot is a program that c an schedule tasks and mark them as done, delete
+ * and save them on a file.
+ *
+ * @author Lim Jin Feng
+ * @version 0.1
+ * @since   2020-08-20
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs Duke Bot in a given file path.
+     *
+     * @param filePath  Where Duke Bot will save it's tasks.
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
```

Comment:
Try not to use `this` unnecessarily. According to the advanced style guide, 
<img width="993" alt="image" src="https://user-images.githubusercontent.com/4933577/92136179-49194f80-ee3e-11ea-9e49-5e90e3d1929b.png">